lib/rules/onClick.js

```diff
@@ -41,19 +41,30 @@ module.exports = {
                 if (bypass) return;
 
                 context.report({
-                  node,
-                  message: getError(errors.onClick.message, testAttribute),
-                  fix: function fix(fixer) {
-                    const { nanoid } = require('nanoid');
-                    const suggestedId = nanoid();
-                    const start = node.range[0] + node.name.name.length;
-                    const end = start + 1;
-
-                    return fixer.insertTextAfterRange(
-                      [start, end],
-                      ' ' + testAttribute + '="' + suggestedId + '"'
-                    );
-                  },
+                    node,
+                    message: getError(errors.onClick.message, testAttribute),
+                    fix: function fix(fixer) {
+                        const { nanoid } = require('nanoid');
+                        const suggestedId = nanoid();
+                        const namePositionEnd = node.name.range[1];
+                        const isSingleLine = node.loc.start.line === node.loc.end.line;
+                        const attributeText = `${ testAttribute }="${ suggestedId }"`;
+                        const start = namePositionEnd;
+                        const end = start + 1;
+
+                        const attributePrefix = isSingleLine
+                            ? ''
+                            : new Array(node.loc.start.column + 2).fill(' ').join('');
```

Comment:
reading your description and realizing this 2 is a hardcoded soft tab indent. do we have visibility into the next line's indent? might be nice if we indented based on the existing indent convention if it's easy

Reply:
I could probably do that.
The other devil's advocate point (against me) is that the attribute indentation / newlining should be handled by a separate lint rule, in which case the test-selectors plugin should just butt out.

Honestly, that's probably the cleaner approach.

Reply:
If I routinely ran prettier after linting, this wouldn't have been an issue. 🤦

Reply:
I ultimately like the idea of adding the fix in the "right place"- I know at least in our world, auto-formats are a separate, distinct step from ESLint corrections, so if it "just worked", that'd feel pretty slick. but I think it'd potentially be worse to indent wrong than to not try to indent at all